### ISSUE TITLE
LIS2DW12TR CS-pin problem

### ISSUE THREAD

**Issue by zed65** [rusefi/rusefi#5928]
When I set the accelerometer cs pin in TS it won't show in console.
I have confirmed it with oscilloscope, the CS pin never go low.

**Comment by rusefillc** [comment]
@zed65 "accelerometerCsPin"?

**Comment by rusefillc** [comment]
```
#if EFI_ONBOARD_MEMS
	initAccelerometer();
#endif
```

```
#if !defined(EFI_ONBOARD_MEMS)
 #define EFI_ONBOARD_MEMS FALSE
#endif
```

**Comment by zed65** [comment]
Yes the accelerometerCsPin, so the problem is what? f407-discovery firmware don't have mems onboard?

**Comment by rusefillc** [comment]
> Yes the accelerometerCsPin, so the problem is what? f407-discovery firmware don't have mems onboard?

yes because https://github.com/rusefi/rusefi/commit/a5c7895528990cd1ca671e9bab2df7e4d1e2b6f2 says ``TODO: add support for LIS302 on discovery board``

Real discovery has LIS302 which we seem not to support. We do not have runtime selection for which driver to use, one has to complier with driver matching hardware?

@dron0gus FYI

**Comment by dron0gus** [comment]
@zed65 I'm a little confused. You are using "f407-discovery" fw on your custom HW.
And you have LIS2DW12TR on this board?
While STM32F407-discovery has LIS302DL or LIS3DSH.
@rusefillc what is correct solution?

**Comment by rusefillc** [comment]
Custom fw matching custom hw is realistic proper solution

**Comment by zed65** [comment]
> @zed65 I'm a little confused. You are using "f407-discovery" fw on your custom HW. And you have LIS2DW12TR on this board? While STM32F407-discovery has LIS302DL or LIS3DSH. @rusefillc what is correct solution?

Yes, that's correct. Custom hardware with f40-discovery and LIS2DW12TR

**Comment by rusefillc** [comment]
looks like "won't fix" territory since we do not plan to make hardware choice at runtime

See https://github.com/rusefi/rusefi/wiki/Custom-Firmware which is the direction for custom hardware not compatible with either with official binaries

**Comment by dron0gus** [comment]
@zed65 see https://github.com/rusefi/rusefi/pull/5930 in case Andrey will not merge.

**Comment by rusefillc** [comment]
> looks like "won't fix" territory since we do not plan to make hardware choice at runtime

I was wrong. @dron0gus has just contributed code with runtime hardware detection!

**Comment by zed65** [comment]
I'm still having trouble with the LIS2DW12, on reset I can see activity on CLK and MOSI for a short moment.
But I never see anything on MISO.

**Comment by dron0gus** [comment]
Do you see CS going low while activity on CLK and MOSI?

**Comment by zed65** [comment]
I was wrong, there are some data on MISO but it don't work anyway.
Does it need to be on a specific SPI-bus (1,2,3)? Can it be shared with more stuff (SD-card, EGT)? 
It's alone on SPI2 on my test setup now.
![LIS2spi](https://github.com/rusefi/rusefi/assets/23425941/c7b32322-f05e-4a62-9212-64cd66da3a21)

**Comment by zed65** [comment]
If I flip to "Clock is high when active" in my analyzer it shows this:
![LIS2spi2](https://github.com/rusefi/rusefi/assets/23425941/097c0165-c3a3-46a0-b1c2-ebc37587d029)

**Comment by dron0gus** [comment]
CLK signal looks strange for me. I count only 9 pulses. While 8+8 are expected. I see two groups of pulses. Can you increase sampling rate of you logic analyzer? 16 MHz or even more?
Also I see no reply on MISO, accel should start driving it after 8 clocks on CLK line.

**Comment by dron0gus** [comment]
Just ordered some China pcb with LIS2DW12. Takes up to 1 month to arrive.

**Comment by zed65** [comment]
Yea it's strange, almost as if the chip isn't soldered correct. I've de-soldered it and even replaced it with a new one (same issue).
Here is a log with the analyzer, don't know if you can open it though.
[LIS2DW.zip](https://github.com/rusefi/rusefi/files/14296023/LIS2DW.zip)
![LIS2spi3](https://github.com/rusefi/rusefi/assets/23425941/11a35375-8a96-4e61-89ff-96e53e4b3490)

**Comment by zed65** [comment]
Progress has been made, turns out that JLC has f-up one of the vias on the board.
I've hacked it together and are now using SPI1.
Now the LIS2DW respond but it's not showing anything in Tunerstudio.
![LIS2spi4](https://github.com/rusefi/rusefi/assets/23425941/8f7717ac-68b1-4fcc-84b8-33b0a938a553)

**Comment by dron0gus** [comment]
This one definetlly looks like reading of 6 bytes from adress 0x28 - output register values.
So at least it passed "who am i" check and started driver.
Can you confirm that you have useSpiImu enabled (under "Experimental 2")

**Comment by zed65** [comment]
Imu what??! Gah, I've seen that and thought it said LMU.
Anyway now I see some values in TS I doubt that they are correct tho.
https://rusefi.com/online/view.php?log=1347

**Comment by dron0gus** [comment]
Looks strange. Did you shake board while taking this log or it was stable?
Do you have anything connected to CAN bus? We have support for CAN accelerometer and both SPI and CAN accel drivers are writing to same log channels... So there may be conflict.

**Comment by zed65** [comment]
I rotated the board while logging. I have nothing connected to CAN, I don't even have CAN read/write enabled.

**Comment by zed65** [comment]
I have a feeling that I do more harm then good with my tests.
Maybe we wait until you get your boards.

**Comment by zed65** [comment]
@dron0gus Did you get your boards from china? Have you had any time to test it?

**Comment by dron0gus** [comment]
@zed65 yes. I received IMU. Actually I have already connected it to disco board. But did not have time to look into issue.
